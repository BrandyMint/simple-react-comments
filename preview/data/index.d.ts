export declare const commentsData: {
    authorUrl: string;
    avatarUrl: string;
    createdAt: Date;
    fullName: string;
    text: string;
}[];
