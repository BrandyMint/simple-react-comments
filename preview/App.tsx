import * as React from 'react';
import CommentsBlock from '../src/index';
import { commentsData } from './data/index';

const styles = {
  fontFamily: 'arial',
  margin: '0 auto',
  marginTop: '50px',
  width: '50%',
};

const cardStyles = {
  borderRadius: '10px',
  boxShadow: '0px 2px 10px rgba(0, 0, 0, 0.2)',
  padding: '30px',
};

export interface State {
  comments: any;
}

class App extends React.Component<{}, State> {
  constructor(props) {
    super(props);
    this.state = {
      comments: commentsData,
    };
  }

  public render() {
    return (
      <div style={styles}>
        <h1>react-simple-comments</h1>
        <div style={cardStyles}>
          <CommentsBlock
            comments={this.state.comments}
            signinUrl={'/signin'}
            isLoggedIn
            reactRouter
            onSubmit={text => {
              if (text.length > 0) {
                this.setState({
                  comments: [
                    ...this.state.comments,
                    {
                      authorUrl: '#',
                      avatarUrl:
                        'https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438',
                      createdAt: new Date(),
                      fullName: 'Alexey Ryabov',
                      text,
                    },
                  ],
                });
                console.log('submit:', text);
              }
            }}
          />
        </div>
      </div>
    );
  }
}

export default App;
