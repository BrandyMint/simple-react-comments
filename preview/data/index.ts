export const commentsData = [
  {
    authorUrl: '#',
    avatarUrl:
      'https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438',
    createdAt: new Date(),
    fullName: 'Alexey Ryabov',
    text: '<h1>hello whats up</h1>',
  },
  {
    authorUrl: '/asdf',
    avatarUrl:
      'https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438',
    createdAt: new Date(),
    fullName: 'Alexey Ryabov',
    text: 'reactomments is awesome!',
  },
];
