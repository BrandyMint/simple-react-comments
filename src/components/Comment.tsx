import * as React from 'react';
import { CommentEntity } from '../models';
import Link from './Link';

export interface Props {
  comment: CommentEntity;
  reactRouter: boolean;
}

const Comment: React.StatelessComponent<Props> = ({ comment, reactRouter }) => {
  const { authorUrl, avatarUrl, text, createdAt, fullName } = comment;
  return (
    <div className="rc-comment">
      <Link reactRouter={reactRouter} href={authorUrl}>
        <div
          className="rc-comment__avatar"
          style={{
            backgroundImage: `url(${avatarUrl})`,
          }}
        />
      </Link>
      <div className="rc-comment__col-right">
        <Link reactRouter={reactRouter} href={authorUrl}>
          <div className="rc-comment__name">{fullName}</div>
        </Link>
        <div className="rc-comment__time">{createdAt.toLocaleDateString()}</div>
        <div className="rc-comment__content">{text}</div>
      </div>
    </div>
  );
};

export default Comment;
