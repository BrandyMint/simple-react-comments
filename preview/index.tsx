import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { BrowserRouter, Route } from 'react-router-dom';
import './css/style.scss';

import App from './App';

ReactDOM.render(
  <BrowserRouter>
    <Route component={App} />
  </BrowserRouter>,
  document.getElementById('root'),
);
