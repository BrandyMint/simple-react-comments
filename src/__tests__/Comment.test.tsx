import React from 'react';
import renderer from 'react-test-renderer';
import { commentsData } from '../../preview/data/';
import Comment from '../components/Comment';

it('snapshot for Comment', () => {
  const component = renderer.create(
    <Comment comment={commentsData[0]} reactRouter={false} />,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
  component.unmount();
});
