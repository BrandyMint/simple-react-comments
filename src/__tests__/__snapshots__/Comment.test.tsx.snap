// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for Comment 1`] = `
<div
  className="rc-comment"
>
  <a
    href="#"
  >
    <div
      className="rc-comment__avatar"
      style={
        Object {
          "backgroundImage": "url(https://cdnb.artstation.com/p/users/avatars/000/126/159/large/582fd86d44a71299b5cc51fe9f580471.jpg?1447075438)",
        }
      }
    />
  </a>
  <div
    className="rc-comment__col-right"
  >
    <a
      href="#"
    >
      <div
        className="rc-comment__name"
      >
        Alexey Ryabov
      </div>
    </a>
    <div
      className="rc-comment__time"
    >
      2018-6-2
    </div>
    <div
      className="rc-comment__content"
    >
      &lt;h1&gt;hello whats up&lt;/h1&gt;
    </div>
  </div>
</div>
`;
